Tasks/Common/azure-arm-rest/azure-arm-common.ts

```diff
@@ -182,6 +182,9 @@ export class ApplicationTokenCredentials {
                 if (response.statusCode == 200) {
                     deferred.resolve(response.body.access_token);
                 }
+                else if(response.statusCode == 401 || response.statusCode == 403) {
```

Comment:
One thing we have noticed recently from azure flows is that if the SPN is created very recently - sometimes this call may fail since SPN is not replicated properly. We were planning to add a retry here anyway. Should you add a retry (may be 3 times, with 3 sec delay) for this.